scripts/common.sh

```diff
@@ -246,3 +246,14 @@ function first_word
 
     echo "$words" | cut -d " " -f 1
 }
+
+function split_on_empty_lines_into_numbered_files
+{
+    local input=""
+    [[ -p /dev/stdin ]] && input="$(cat -)"
+
+    path_prefix="${1}"
+    path_suffix="${2}"
+
+    echo "${input}" | awk -v RS= "{print > (\"${path_prefix}_\"NR \"${path_suffix}\")}"
```

Comment:
```suggestion
    path_prefix="${1}"
    path_suffix="${2}"
    awk -v RS= "{print > (\"${path_prefix}_\"NR \"${path_suffix}\")}"
```
Won't this just work?

Reply:
@ekpyron I can see one problem with a suggested solution. It gets stuck in case of an empty stdin. To avoid that we can keep current code or have a failing assertion for an empty stdin. I have no strong feelings on which solution is better.

Reply:
You mean just plainly using ``split_on_empty_lines_into_numbered_files X Y`` gets stuck? I wouldn't say that's a huge problem. And even ``echo -n "" | split_on_empty_lines_into_numbered_files X Y`` should still work, shouldn't it?

Reply:
Yes, plainly used `split_on_empty_lines_into_numbered_files X Y` gets stuck and bash is hanging without any user notification. And, I can see the lack of notification to be a little problem. That's because, it can lead to CI fails with only timeout information, nothing more. However, if you still prefer a simplified version, I can do an update :)
`echo -n "" | split_on_empty_lines_into_numbered_files X Y` works just fine.

Reply:
I'd prefer the simpler version :-).
I don't think it's an issue if a new PR using this function incorrectly fails CI - it's not like it'll cause random CI failures. You can just add a comment to the function that states that you have to pipe something to it.

Reply:
Code updated :)